FirebaseAppCheck/Sources/Core/FIRAppCheck.m

```diff
@@ -325,12 +325,19 @@ - (nonnull NSString *)notificationTokenKey {
         // TODO: Make sure the self.tokenRefresher is updated only once. Currently the timer will be
         // updated twice in the case when the refresh triggered by self.tokenRefresher, but it
         // should be fine for now as it is a relatively cheap operation.
-        __auto_type refreshResult = [[FIRAppCheckTokenRefreshResult alloc]
-            initWithStatusSuccessAndExpirationDate:token.expirationDate
-                                    receivedAtDate:token.receivedAtDate];
-        [self.tokenRefresher updateWithRefreshResult:refreshResult];
-        [self postTokenUpdateNotificationWithToken:token];
-        return token;
+        if (token) {
```

Comment:
Maybe I've been doing Swift too long, but why are lines 321 and 324 `Nullable`?

Should null be detected earlier and pass along nonnulls?

Reply:
I think you're right, but I suspect that there may have been a case where line 321 was justified. In that case, setting a `nil` token in storage would wipe storage– which may be desireable.

I don't however think that line 324 should use a non-nil token.

Reply:
I'm not sure that line 324 should be non-nil. `[self.storage setToken:token]` on line 322 resolves the promise with `nil` unless there's an error removing the token from storage [here](https://github.com/firebase/firebase-ios-sdk/blob/d89d65e52854f6629614e2d7ec4608d706d4bd2b/FirebaseAppCheck/Sources/Core/Storage/FIRAppCheckStorage.m#L92-L94).